### PR TITLE
add an option to source password from file

### PR DESCRIPTION
**Pull request by aneesh-joseph**
Adding an option to source the git password from a file . This will help in cases where the password can be injected as a file using say Hashicorp vault injector

### CODE REVIEW

File: cmd/git-sync/main.go [hunk 1]
```diff
@@ -85,7 +85,8 @@ var flUsername = flag.String("username", envString("GIT_SYNC_USERNAME", ""),
 	"the username to use for git auth")
 var flPassword = flag.String("password", envString("GIT_SYNC_PASSWORD", ""),
 	"the password to use for git auth (users should prefer env vars for passwords)")
-
+var flPasswordFile = flag.String("password-file", envString("GIT_SYNC_PASSWORD_FILE", ""),
+	"the file from which password for git auth has to be sourced")
```
Comment (thockin) [inline cmd/git-sync/main.go:89]: nit: "the file from which the password for git auth will be sourced"

File: cmd/git-sync/main.go [hunk 2]
```diff
@@ -238,8 +239,8 @@ func main() {
 		os.Exit(1)
 	}
 
-	if (*flUsername != "" || *flPassword != "" || *flCookieFile || *flAskPassURL != "") && *flSSH {
-		fmt.Fprintf(os.Stderr, "ERROR: --ssh is set but --username, --password, --askpass-url, or --cookie-file were provided\n")
+	if (*flUsername != "" || *flPassword != "" || *flPasswordFile != "" || *flCookieFile || *flAskPassURL != "") && *flSSH {
+		fmt.Fprintf(os.Stderr, "ERROR: --ssh is set but --username, --password, --password-file, --askpass-url, or --cookie-file were provided\n")
 		os.Exit(1)
 	}
```
Comment (thockin) [inline cmd/git-sync/main.go:245]: Probably need a block to ensure that user doesn't set both flPassword and flPasswordFile at the same time

Comment (thockin) [reply cmd/git-sync/main.go:245]: maybe also a block to check that one of flPassword and flPasswordFile are set when flUsername is set ?

File: cmd/git-sync/main.go [hunk 3]
```diff
@@ -254,7 +255,14 @@ func main() {
 	// `git clone`, so initTimeout set to 30 seconds should be enough.
 	ctx, cancel := context.WithTimeout(context.Background(), initTimeout)
 
-	if *flUsername != "" && *flPassword != "" {
+	if *flUsername != "" && ( *flPassword != "" ||  *flPasswordFile != "") {
+		if *flPasswordFile != "" {
+			flPassword, err := ioutil.ReadFile(*flPasswordFile)
```
Comment (thockin) [inline cmd/git-sync/main.go:260]: This doesn't do what you want - flPassword will go out of scope immediately.

**Comment by thockin** [comment]
Hi, is this PR still something you want to work on?

**Review by lastthyme — APPROVED** [review]

**Comment by zhouhaibing089** [comment]
@aneesh-joseph: Hi Annesh, are you still interested on working on this? Hopefully you don't mind if I take it up~

**Comment by zhouhaibing089** [comment]
Opened #431 as a continuation of this change.

**Comment by stp-ip** [comment]
Closing as this was continued and merged in #431 Thanks to everyone involved \o/

**Pull request closed without merging**